pkgs/by-name/ht/http3-ytproxy/package.nix

```diff
@@ -0,0 +1,49 @@
+{ lib
+, fetchFromGitHub
+, buildGoModule
+, libwebp
+}:
+buildGoModule rec {
+  pname = "http3-ytproxy";
+  version = "unstable-2022-07-03";
+
+  src = fetchFromGitHub {
+    owner = "TeamPiped";
+    repo = pname;
+    rev = "4059da180bb9d7b0de10c1a041bd0e134f1b6408";
+    hash = "sha256-ilIOkZ9lcuSigh/mMU7IGpWlFMFb2/Y11ri3659S8+I=";
+  };
+
+  postPatch = ''
+    substituteInPlace main.go \
+      --replace 'lucas-clemente/quic-go' 'quic-go/quic-go'
+  '';
+
+  overrideModAttrs = _: {
+    preBuild = ''
+      substituteInPlace go.mod \
+        --replace 'go 1.16' 'go 1.20'
+      go mod tidy
+    '';
+    postInstall = ''
+      cp go.mod "$out/go.mod"
+    '';
+  };
```

Comment:
Why do we need to bump the go version in go.mod? Normally this shouldn't be required. Also I think it would be easier if do this in postPatch which should also avoid the preBuild phase.